src/bootstrap/config.rs

```diff
@@ -882,25 +888,15 @@ impl Config {
                 check_ci_llvm!(llvm.build_config);
                 check_ci_llvm!(llvm.plugins);
 
-                // CI-built LLVM can be either dynamic or static.
-                let ci_llvm = config.out.join(&*config.build.triple).join("ci-llvm");
-                config.llvm_link_shared = if config.dry_run {
-                    // just assume dynamic for now
-                    true
-                } else {
-                    let link_type = t!(
-                        std::fs::read_to_string(ci_llvm.join("link-type.txt")),
-                        format!("CI llvm missing: {}", ci_llvm.display())
-                    );
-                    link_type == "dynamic"
-                };
+                // CI-built LLVM can be either dynamic or static. We won't know until we download it.
+                config.llvm_link_shared.set(None);
```

Comment:
set(None) seems like a no-op -- I think a better option here is probably to move this comment up to the check_ci_llvm line for llvm_link_shared.